### ISSUE TITLE
Impossible to retrieve attachments

### ISSUE THREAD

**Issue by jfthuong** [RallyTools/RallyRestToolkitForPython#138]
Hi,

In the function `getAttachment` (_restapi.py_), the commit 4f09791983c6b721aab3142a101fb08a64af7c74 replaced `base64.decodestring` by `base64.decodebyte`; this leads to 2 issues:

1. This code is not compatible anymore with Python 2.7
2. The content of the attachment is stored as a string and shall be converted to bytes before being used: `base64.decodebytes(att_content.Content.encode('ascii'))`

It is currently impossible to retrieve an attachment with the function `getAttachment`~! 😢

**Comment by jfthuong** [comment]
A suggestion of modification would be

```python
                content = att_content.Content
                if sys.version_info >= (3, 0, 0):
                    att.Content = base64.decodebytes(content.encode('ascii'))
                else:
                    att.Content = base64.decodestring(content)
```

**Comment by sc305495** [comment]
I've also run into a similar issue w/ Python 3.6, i've created a post on SO https://stackoverflow.com/questions/49984802/pyral-getattachment/49988876#49988876

**Comment by jfthuong** [comment]
The answer to this post on SO provdides a better solution `base64.b64decode`.

**Comment by geekychandraul** [comment]
import sys
import string
import base64
from pyral import rallyWorkset, Rally,RallyRESTResponse

rally = Rally(server, user=USER_NAME, password=PASSWORD, workspace=workspace, project=project)
criterion = 'FormattedID = US57844'
response = rally.get('HierarchicalRequirement',  query=criterion, order="FormattedID",pagesize=200, limit=400, projectScopeDown=True)

artifact = response.next()
context, augments = rally.contextHelper.identifyContext()
for att in artifact.Attachments:
    resp = rally._getResourceByOID(context, 'AttachmentContent', att.Content.oid, project=None)
    if resp.status_code not in [200, 201, 202]:
        break
    res = RallyRESTResponse(rally.session, context, "AttachmentContent.x", resp, "full", 1)
    if res.errors or res.resultCount != 1:
        print("breaking the for loop")
    att_content = res.next()
    cont = att_content.Content
    x = base64.b64decode(cont)
    output = open(att.Name, 'wb') 
    output.write(x)

**Comment by geekychandraul** [comment]
Hope this helps.

`import sys
import string
import base64
from pyral import rallyWorkset, Rally,RallyRESTResponse

rally = Rally(server, user=USER_NAME, password=PASSWORD, workspace=workspace, project=project)
criterion = 'FormattedID = US57844'
response = rally.get('HierarchicalRequirement',  query=criterion, order="FormattedID",pagesize=200, limit=400, projectScopeDown=True)

artifact = response.next()
context, augments = rally.contextHelper.identifyContext()
for att in artifact.Attachments:
    resp = rally._getResourceByOID(context, 'AttachmentContent', att.Content.oid, project=None)
    if resp.status_code not in [200, 201, 202]:
        break
    res = RallyRESTResponse(rally.session, context, "AttachmentContent.x", resp, "full", 1)
    if res.errors or res.resultCount != 1:
        print("breaking the for loop")
    att_content = res.next()
    cont = att_content.Content
    x = base64.b64decode(cont)
    output = open(att.Name, 'wb') 
    output.write(x)`

**Comment by geekychandraul** [comment]
Hope this helps.

```
import sys
import string
import base64
from pyral import rallyWorkset, Rally,RallyRESTResponse

rally = Rally(server, user=USER_NAME, password=PASSWORD, workspace=workspace, project=project)
criterion = 'FormattedID = US57844'
response = rally.get('HierarchicalRequirement',  query=criterion, order="FormattedID",pagesize=200, limit=400, projectScopeDown=True)

artifact = response.next()
context, augments = rally.contextHelper.identifyContext()
for att in artifact.Attachments:
    resp = rally._getResourceByOID(context, 'AttachmentContent', att.Content.oid, project=None)
    if resp.status_code not in [200, 201, 202]:
        break
    res = RallyRESTResponse(rally.session, context, "AttachmentContent.x", resp, "full", 1)
    if res.errors or res.resultCount != 1:
        print("breaking the for loop")
    att_content = res.next()
    cont = att_content.Content
    x = base64.b64decode(cont)
    output = open(att.Name, 'wb') 
    output.write(x)
```

**Comment by klehman-rally** [comment]
This has been fixed in the 1.5.0 release.
